File: Java/pattern_printing/q5.java
//           1 
//         2 1 2 
//       3 2 1 2 3 
//     4 3 2 1 2 3 4 
//   5 4 3 2 1 2 3 4 5 
// 6 5 4 3 2 1 2 3 4 5 6 

// Java Program to print pattern
// Palindrome triangle
import java.util.*;

public class pattern {
    
    // Function to demonstrate pattern
    public static void printPattern(int n)
    {
        int i, j;

        // outer loop to handle number of rows
        for (i = 1; i <= n; i++) {
            
            // inner loop to print the spaces
            for (j = 1; j <= 2 * (n - i); j++) {
                System.out.print(" ");
            }

            // inner loop to print the first part
            for (j = i; j >= 1; j--) {
                System.out.print(j + " ");
            }

            // inner loop to print the second part
            for (j = 2; j <= i; j++) {
                System.out.print(j + " ");
            }

            // printing new line for each row
            System.out.println();
        }
    }

    // Driver Function
    public static void main(String args[])
    {
        int n = 6;
        printPattern(n);
    }
}

File: C/pattern_printing/q5.c
// * * * * *    |    1 2 3 4 5    |    A B C D E
//   * * * *    |      1 2 3 4    |      A B C D
//     * * *    |        1 2 3    |        A B C
//       * *    |          1 2    |          A B
//         *    |            1    |            A

#include <stdio.h>

int main()
{
    int rows = 5;

    // Outer loop for printing all rows
    for (int i = 0; i < rows; i++)
    {

        // First Inner loop for printing white spaces
        for (int j = 0; j < 2 * i; j++)
        {
            printf(" ");
        }

        // Second inner loop for printing star *
        for (int k = 0; k < rows - i; k++)
        {
            printf("* ");
        }
        printf("\n");
    }

    return 0;
}
//This pattern is the 180° flipped version of the left half pyramid pattern we discussed earlier.